libs/ui/components/inputs/button/props.ts

```diff
@@ -1,14 +1,15 @@
 import { ButtonProps, StackDirection } from "@chakra-ui/react";
-
 export default interface IButtonProps extends ButtonProps {
-    startEnhancer?: React.ReactElement,
-    startEnhancerHover?: React.ReactElement,
-    endEnhancer?: React.ReactElement,
-    endEnhancerHover?: React.ReactElement,
+    startEnhancer?: (enhancerProps: EnhancerProps) => React.ReactElement,
+    endEnhancer?: (enhancerProps: EnhancerProps) => React.ReactElement,
     hoverColor?: string,
     hoverBackgroundColor?: string,
     direction?: StackDirection,
     spacing?: string,
     children?: React.ReactElement | string;
     color?: string;
 }
+
+export type EnhancerProps = {
+    isHovered: boolean | undefined;
```

Comment:
```suggestion
    isHovered?: boolean;
```

With the ?: it will be undefined if you don't specify it in the props